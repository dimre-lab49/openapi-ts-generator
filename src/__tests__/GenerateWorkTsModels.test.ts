import { existsSync, readdirSync, readFileSync } from 'fs';
import { generateTsModels } from '../index';

beforeAll(async () => {
  await generateTsModels(
    'https://d-work-wal-00-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
    './output_wrk/',
  );
});

test('Generate TS Models should generate an output directory', async done => {
  try {
    const files = await readdirSync('./output_wrk');
    expect(files.length).toBeGreaterThan(0);
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('Generate TS Models should generate an output directory', async done => {
  try {
    const files = await readdirSync('./output_wrk');
    expect(files.length).toBeGreaterThan(0);
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('Generate TS Models should generate an index.ts barrel file', async done => {
  try {
    const fileExist = await existsSync('./output_wrk/index.ts');
    expect(fileExist).toBeTruthy();
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('IAggregatedWorkModel should be correct', async done => {
  try {
    const file = await readFileSync('./output_wrk/aggregated-work-model.model.ts');
    expect(file.toString()).toMatchSnapshot();
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('AggregatedWorkModelFormGroupFac should be correct', async done => {
  try {
    const file = await readFileSync('./output_wrk/aggregated-work-model.form-group-fac.ts');
    expect(file.toString()).toMatchSnapshot();
    done();
  } catch (err) {
    done.fail(err);
  }
});
