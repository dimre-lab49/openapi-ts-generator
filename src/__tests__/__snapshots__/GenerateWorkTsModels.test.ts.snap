// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IAggregatedWorkModel should be correct 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-model-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
 // tslint:disable: array-type
 
import { DayOfWeek } from './day-of-week.enum';

export interface IAggregatedWorkModel {
    createdBy?: string;
    createdOnUtc?: Date;
    dayOfWeek?: DayOfWeek;
    id?: string;
    metric?: number;
    modelName?: string;
    siteCollectionId?: string;
    siteCollectionNumber?: string;
    tenantId?: string;
    updatedBy?: string;
    updatedOnUtc?: Date;
    weekOfYear?: number;
    workDate?: Date;
    workType?: string;
    year?: number;
}

export enum AggregatedWorkModelProperties {
    CREATED_BY = 'createdBy',
    CREATED_ON_UTC = 'createdOnUtc',
    DAY_OF_WEEK = 'dayOfWeek',
    ID = 'id',
    METRIC = 'metric',
    MODEL_NAME = 'modelName',
    SITE_COLLECTION_ID = 'siteCollectionId',
    SITE_COLLECTION_NUMBER = 'siteCollectionNumber',
    TENANT_ID = 'tenantId',
    UPDATED_BY = 'updatedBy',
    UPDATED_ON_UTC = 'updatedOnUtc',
    WEEK_OF_YEAR = 'weekOfYear',
    WORK_DATE = 'workDate',
    WORK_TYPE = 'workType',
    YEAR = 'year',
}
"
`;
